docs/site/migration/datasources.md

```diff
@@ -6,8 +6,72 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/migration-datasources.html
 ---
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3946](https://github.com/strongloop/loopback-next/issues/3946).
-" %}
+## Overview
+
+LoopBack 3 datasources are compatible with LoopBack 4 datasources, so migrating
+datasources from LoopBack 3 to LoopBack 4 is simple.
+
+In LoopBack 3, all datasources are defined in the `server/datasources.json`
+file, whereas in LoopBack 4 each datasource is defined in its own file in the
+`src/datasources` folder. Each LoopBack 4 datasource has a configuration file
+(e.g. `db.datasource.config.ts`) and a class file (e.g. `db.datasource.ts`).
+
+## Migration Steps
+
+To migrate a datasource from LoopBack 3 to LoopBack 4, complete the following
+steps:
+
+1. In the root of your LoopBack 4 application, use the `lb4 datasource` command
+   to create a new datasource and enter the same datasource name as your
+   LoopBack 3 application's datasource (e.g. `db`):
+
+   ```
+   $ lb4 datasource
+   ? Datasource name: db
+   ```
+
+2. For the remaining prompts from the `lb4 datasource` command, use the defaults
+   (press Enter for each one) since these will be replaced in the next step:
+
+   ```
+   ? Select the connector for db: In-memory db (supported by StrongLoop)
+   ? window.localStorage key to use for persistence (browser only):
+   ? Full path to file for persistence (server only):
+   ```
+
+3. Replace the contents of the newly created
+   `src/datasources/{dataSource.dataSourceName}.datasource.config.ts` file in
+   your LoopBack 4 application with the datasource configuration from
+   `server/datasources.json` in your LoopBack 3 application.
+
+   {% include code-caption.html content="server/datasources.json" %}
+
+   ```json
+   {
+     "db": {
+       "name": "db",
+       "connector": "memory"
```

Comment:
Do we have any config such as `  "file": "./data/db.json"` in LB3? I found it weird that we set the above prompts to empty, but we don't replace those setting with any configuration in this step. It would be good if we can make the example more complete.

Reply:
Updated it to use `mysql` so it's clearer 👍